In [ ]:
pip install requests beautifulsoup4 pandas

In [ ]:
pip install google

In [ ]:
import csv

In [ ]:
li=[]
try:
	from googlesearch import search
except ImportError:
	print("No module named 'google' found")

# to search
query = "industry in which Canoo operates"

for j in search(query, tld="co.in", num=20, stop=20, pause=2):
	li.append(j)

for i in li:
  print (i)



https://en.wikipedia.org/wiki/Canoo
https://www.canoo.com/about/
https://www.canoo.com/
https://investors.canoo.com/company-information
https://www.globaldata.com/company-profile/canoo-inc/
https://www.industryweek.com/leadership/companies-executives/article/21263239/canoo-sticks-to-2023-production-target
https://www.press.canoo.com/press-release/canoo-announces-ev-battery-module-manufacturing-facility-in-oklahoma
https://www.prnewswire.com/news-releases/canoo-delivering-first-made-in-oklahoma-electric-vehicles-to-state-government-301985717.html
https://investors.canoo.com/
https://www.bloomberg.com/profile/company/GOEV:US
https://tulsaworld.com/news/local/business/canoos-proposed-electric-vehicle-plant-in-pryor-could-be-delayed-industrial-park-head-says/article_4421c660-f896-11ec-945c-93628edd01aa.html
https://www.linkedin.com/company/canoo
https://www.designdevelopmenttoday.com/industries/automotive/news/22879105/canoo-begins-shipping-evs-from-its-oklahoma-factory
https://readwrite.c

In [ ]:
import requests as req

from bs4 import BeautifulSoup
import pandas

In [ ]:
data=[]
for i in li:
  res=req.get(i)
  # print(i,res)
  soup = BeautifulSoup(res.content, "html.parser")
# Extract the text content of the webpage
  txt = soup.get_text()
# print(text)
  # text=soup.find(string='canoo')
  # print(txt)





In [ ]:
!pip install newspaper3k
!pip install bs4
!pip install readability

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 211.1/211.1 kB 4.5 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 81.3/81.3 kB 9.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.7/97.7 kB 11.5 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.4/7.4 MB 20.1 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Preparing metadata (setup.py) ... done
  Preparing metadata (setup.py) ... done
  Created wheel for tinysegmenter: filename=tinysegmenter-0.3-py3-none-any.whl size=13538 sha256=e075b7eaf7d792fd1b47b1c286c73bfc6ab412ace3e24f29dc69268679f26fda
  Stored in directory: /root/.cache/pip/wheels/c8/d6/6c/384f58df48c00b9a31d638005143b5b3ac62c3d25fb1447f23
  Created wheel for feedfinder2: filename=feedfinder2-0.0.4-py3-none-any.whl size=3340 sha256=bb3df5b41b7299d614a49507ea5aabc3ce0f6843522645c4bf33379865503ef1
  Stored in directory: /root/.cache/pip/wheels/97/02/e7/a1ff1760e12bdbaab0ac824fae5c1b

In [ ]:
import nltk
nltk.download('punkt')
from newspaper import Article
import requests

# from readability.readability import Document as Paper
from requests.packages.urllib3.exceptions import InsecureRequestWarning
import re

[nltk_data] Downloading package punkt to /root/nltk_data...
[nltk_data]   Unzipping tokenizers/punkt.zip.


In [ ]:
import itertools
li1=[]
for i in li:
  url=i
  article=Article(url)
  try:
    article.download()
    article.parse()
    article.text
    article.nlp()

  except:
    pass
  print(article.keywords)
  temp=article.keywords
  li1.append(temp)
filename = "company.csv"
with open(filename, 'w') as csvfile:
    # creating a csv writer object
  csvwriter = csv.writer(csvfile)
  csvwriter.writerows(list(itertools.zip_longest(*li1, fillvalue='')))



['company', '2021', 'canoo', 'wikipedia', 'production', 'vehicles', 'announced', '2022', 'electric', 'vehicle', 'plant']
['company', 'lifecycle', 'texas', 'vehicledistinguished', 'automotive', 'wide', 'leading', 'uscanoos', 'technologies', 'vehicle', 'unique', 'technology']
[]
['company', 'canoo', 'information', 'automotive', 'wide', 'vehicles', 'traditional', 'technologies', 'electric', 'vehicle', 'unique', 'technology']
['company', 'canoo', 'informed', 'businessgain', 'overview', 'decisions', 'gain', '360degree', 'view', 'profile', 'business']
['company', 'canoo', 'q4', 'production', '2023', 'vans', 'million', 'sticks', 'executives', 'vehicle', 'aquila', 'target', 'trading', 'rd']
['module', 'canoo', 'midamerica', 'oklahoma', 'pryor', 'forwardlooking', 'battery', 'facility', 'park', 'canoos', 'industrial', 'manufacturing', 'ev', 'statements', 'press', 'future']
['canoo', 'oklahoma', 'events', 'forwardlooking', 'vehicles', 'canoos', 'electric', 'state', 'statements', 'vehicle', 'deliv

In [ ]:
import csv

# opening the CSV file
with open('company.csv', mode ='r')as file:

  # reading the CSV file
  csvFile = csv.reader(file)

  # displaying the contents of the CSV file
  for lines in csvFile:
        print(lines)

['company', 'company', '', 'company', 'company', 'company', 'module', 'canoo', 'automotive', 'company', '2021', '2024', 'shipping', 'company', '', 'nation', 'company', 'company', 'company', 'canoo']
['2021', 'lifecycle', '', 'canoo', 'canoo', 'canoo', 'canoo', 'oklahoma', 'wide', 'canoo', 'stewart', 'canoo', 'canoo', 'canoo', '', 'partner', 'canoo', 'canoo', 'canoo', 'production']
['canoo', 'texas', '', 'information', 'informed', 'q4', 'midamerica', 'events', 'vehicles', 'speak', 'oklahoma', 'oklahoma', 'factory', 'thats', '', 'canoo', 'know', 'commerce', 'buys', 'vehicles']
['wikipedia', 'vehicledistinguished', '', 'automotive', 'businessgain', 'production', 'oklahoma', 'forwardlooking', 'relations', 'need', 'million', 'linkedin', 'oklahoma', 'wrong', '', 'energy', 'production', 'oklahoma', 'pryor', 'electric']
['production', 'automotive', '', 'wide', 'overview', '2023', 'pryor', 'vehicles', 'traditional', 'todaymarkets', 'industrial', 'vehicles', 'evs', 'federal', '', 'opportunities'

**CONCLUSION- CANOOS IS ELECTRIC VEHICLE COMPANY**

In [ ]:
import pandas as pd
x=pd.read_csv('company.csv')
x.to_csv('company.csv')

In [ ]:

li=[]
try:
	from googlesearch import search
except ImportError:
	print("No module named 'google' found")

# to search
query = "financial statement and growth of Canoos EV company"

for j in search(query, tld="co.in", num=20, stop=20, pause=2):
	li.append(j)

for i in li:
  print (i)

https://www.wsj.com/market-data/quotes/GOEV/financials
https://investors.canoo.com/
https://investors.canoo.com/sec-filings/annual-reports
https://investors.canoo.com/financial-information/income-statement
https://investors.canoo.com/financial-information/balance-sheet
https://investors.canoo.com/financial-information/financial-results
https://www.macroaxis.com/financials/GOEV/Canoo-Inc
https://finance.yahoo.com/quote/GOEV/key-statistics/
https://www.sec.gov/Archives/edgar/data/1750153/000155837021003617/tmb-20210329xex99d1.htm
https://www.press.canoo.com/press-release/canoo-inc-announces-third-quarter-2023-results
https://www.gurufocus.com/news/2124109/canoo-inc-goev-reports-third-quarter-2023-results-mixed-financial-performance-amidst-expansion-efforts
https://finance.yahoo.com/news/canoo-inc-goev-reports-third-013523519.html
https://fullratio.com/stocks/nasdaq-goev/canoo
https://stockanalysis.com/stocks/goev/statistics/
https://investors.canoo.com/news-presentations/press-releases/d

In [ ]:
li2=[]
for i in li:
  url=i
  article=Article(url)

  try:
    article.download()
    article.parse()
    article.text
    article.nlp()

  except:
    pass
  print(article.keywords)
  temp=article.keywords
  li2.append(temp)

filename = "canoo_finance.csv"
with open(filename, 'w') as csvfile:
    # creating a csv writer object
  csvwriter = csv.writer(csvfile)
  csvwriter.writerows(li2)
  csvwriter.writerows(list(itertools.zip_longest(*li2, fillvalue='')))





[]
['automotive', 'wide', 'vehicles', 'relations', 'traditional', 'team', 'technologies', 'investor', 'electric', 'vehicle', 'unique', 'technology']
['annual', 'reports']
['type', 'significant', 'costs', '1publisher', 'knowledge', 'income', 'accounting', 'process', 'fasburi', 'statement', 'product', 'standards']
['type', 's99paragraph', 'resources', 'sheet', 'terms', 'sx', 'fasburi', 'accounting', 'balance', 'occur', '1subparagraph', 'standards']
['results', 'financial']
['company', 'price', 'investors', 'canoo', 'today', 'canoos', 'financial', 'financials', 'determine', 'companys', 'value', 'market', 'future']
['statistics', 'historical', 'canoo', 'plus', 'access', 'financial', 'valuation', '40', 'data', 'goev', 'essentiallearn', 'finance', 'measures']
['canoo', 'results', 'measure', 'loss', 'ebitda', 'net', 'performance', 'adjusted', 'measures', 'gaap']
['announces', '30', 'compared', 'results', 'canoo', 'loss', 'ended', 'months', '2023', 'million', 'quarter', 'net', 'cash', 'adjuste

In [ ]:
import csv

# opening the CSV file
with open('canoo_finance.csv', mode ='r')as file:

  # reading the CSV file
  csvFile = csv.reader(file)

  # displaying the contents of the CSV file
  for lines in csvFile:
    print(lines)



[]
['automotive', 'wide', 'vehicles', 'relations', 'traditional', 'team', 'technologies', 'investor', 'electric', 'vehicle', 'unique', 'technology']
['annual', 'reports']
['type', 'significant', 'costs', '1publisher', 'knowledge', 'income', 'accounting', 'process', 'fasburi', 'statement', 'product', 'standards']
['type', 's99paragraph', 'resources', 'sheet', 'terms', 'sx', 'fasburi', 'accounting', 'balance', 'occur', '1subparagraph', 'standards']
['results', 'financial']
['company', 'price', 'investors', 'canoo', 'today', 'canoos', 'financial', 'financials', 'determine', 'companys', 'value', 'market', 'future']
['statistics', 'historical', 'canoo', 'plus', 'access', 'financial', 'valuation', '40', 'data', 'goev', 'essentiallearn', 'finance', 'measures']
['canoo', 'results', 'measure', 'loss', 'ebitda', 'net', 'performance', 'adjusted', 'measures', 'gaap']
['announces', '30', 'compared', 'results', 'canoo', 'loss', 'ended', 'months', '2023', 'million', 'quarter', 'net', 'cash', 'adjuste

In [ ]:
x=pd.read_csv('canoo_finance.csv')
x.to_csv('canoo_finance.csv')

ParserError: Error tokenizing data. C error: Expected 12 fields in line 7, saw 13


In [ ]:
url=li[9]
article=Article(url)
article.download()
article.parse()
article.text
article.nlp()
print(article.summary)


Justin, TX – (November 14, 2023) – Canoo Technologies (NASDAQ: GOEV), a leading high-tech advanced mobility company, today announced its financial results for the third quarter of 2023.
• GAAP Net Loss per share of $(0.18) and $(0.53) for the three and nine months ended September 30, 2023, compared to $(0.43) and $(1.62) for the three and nine months ended September 30, 2022.
• Adjusted EPS per share of $(0.07) and $(0.36) for the three and nine months ended September 30, 2023, compared to $(0.31) and $(1.43) for the three and nine months ended September 30, 2022.
• Net cash used in operating activities totaled $191.4 million for the nine months ended September 30, 2023, compared to $329.9 million for the nine months ended September 30, 2022.
• Net cash used in investing activities was $45.4 million during the nine months ended September 30, 2023, compared to $58.4 million net cash used in investing activities during the nine months ended September 30, 2022.


CANOO'S FINANCIAL STATEMENT -

Justin, TX – (November 14, 2023) – Canoo Technologies (NASDAQ: GOEV), a leading high-tech advanced mobility company, today announced its financial results for the third quarter of 2023.
• GAAP Net Loss per share of $(0.18) and $(0.53) for the three and nine months ended September 30, 2023, compared to $(0.43) and $(1.62) for the three and nine months ended September 30, 2022.
• Adjusted EPS per share of $(0.07) and $(0.36) for the three and nine months ended September 30, 2023, compared to $(0.31) and $(1.43) for the three and nine months ended September 30, 2022.
• Net cash used in operating activities totaled $191.4 million for the nine months ended September 30, 2023, compared to $329.9 million for the nine months ended September 30, 2022.
• Net cash used in investing activities was $45.4 million during the nine months ended September 30, 2023, compared to $58.4 million net cash used in investing activities during the nine months ended September 30, 2022.

In [ ]:
li=[]
try:
	from googlesearch import search
except ImportError:
	print("No module named 'google' found")

# to search
query = "key players in EV"

for j in search(query, tld="co.in", num=20, stop=20, pause=2):
	li.append(j)

for i in li:
  print (i)

https://ts2.com.pl/en/who-are-the-key-players-in-ev/
https://www.marketsandmarkets.com/Market-Reports/electric-vehicle-market-209371461.html
https://www.mordorintelligence.com/industry-reports/electric-vehicle-market
https://capital.com/top-20-global-electric-vehicle-companies-ranking-tesla-ev-top-dog
https://abnresource.com/22/abn-news/key-players-electric-vehicle-revolution/
https://www.fool.com/research/largest-ev-companies/
https://www.fortunebusinessinsights.com/industry-reports/electric-vehicle-market-101678
https://companiesmarketcap.com/electric-vehicles/largest-ev-companies-by-market-cap/
https://genuineimpact.substack.com/p/mapping-out-key-players-in-evs
https://energydigital.com/top10/top-10-electric-vehicle-companies
https://www.linkedin.com/pulse/ev-phev-market-insights-players-forecast-2030-reportprime
https://www.imarcgroup.com/electric-vehicles-manufacturers-companies
https://finance.yahoo.com/news/global-ev-market-share-company-124904822.html
https://www.globaldata.com

In [ ]:
for i in li:
  url=i
  article=Article(url)

  try:
    article.download()
    article.parse()
    article.text
    article.nlp()

  except:
    pass
  print(article.keywords)

['players', 'group', 'electric', 'industry', 'market', 'key', 'sustainable', 'tesla', 'vehicle', 'ev']
['2030', 'size', 'electric', 'market', 'charging', 'high', 'evs', 'share', 'batteries', 'forecast', 'usd', 'vehicle', 'report', 'ev', 'vehicles']
['data', 'size', 'electric', 'market', 'industry', 'growth', 'share', 'spreadsheets', 'significant', 'vehicle', 'vehicles', 'analysis']
['biggest', 'electric', 'market', 'firms', 'car', '2022', 'li', 'vehicle', 'ev', 'vehicles', 'companies']
['company', 'plans', 'recently', 'players', 'electric', 'charging', 'key', 'evs', 'revolution', 'points', 'announced', 'vehicle', 'ev']
['company', 'phevs', 'largest', 'motors', 'evs', 'sales', 'manufactured', '2022', '2023', 'bevs', 'ev', 'founded', 'companies']
['segment', 'drive', 'size', 'electric', 'market', 'evs', 'growth', 'share', 'vehicle', 'ev', 'vehicles']
['list', 'trucks', 'speciality', 'largest', 'electric', 'cap', 'market', 'stock', 'shown', 'manufacture', 'vehicle', 'ev', 'vehicles', 'com

**CONCLUSION - TESLA IS LEADING COMPANY IN THIS DOMAIN.
VARIOUS COMPANY LIKE TOYOTA, VOLVO AND MANY ARE WORKING IN SAME DOMAIN.
The global EV market is projected to reach 434.60 billion units by 2030.

**

In [ ]:
rl=li[2]
article=Article(url)
article.download()
article.parse()
print(article.text)
article.nlp()
print(article.summary)

EV Market 2022-2030:

The global electric vehicle market was valued at $163.01 billion in 2020, and is projected to reach $823.75 billion by 2030, registering a CAGR of 18.2% from 2021 to 2030.

An electric vehicle operates on electricity unlike its counterpart, which runs on fuel. Instead of internal combustion engine, these vehicles run on an electric motor that requires constant supply of energy from batteries to operate. There are a variety of batteries used in these vehicles. These include lithium ion, molten salt, zinc-air, and various nickel-based designs. The electric vehicle was primarily designed to replace conventional ways of travel as they lead to environmental pollution. It has gained popularity, owing to numerous technological advancements. It outperforms conventional vehicles providing higher fuel economy, low carbon emission & maintenance, convenience of charging at home, smoother drive, and reduced sound from engine. There are three types of electric vehicles batterie